Source/Dafny/Resolver.cs

```diff
@@ -11134,10 +11102,12 @@ private class RBranchExpr : RBranch {
       List<LiteralExpr> alternatives = new List<LiteralExpr>();
       foreach (var PB in pairPB) {
         var pat = PB.Item1;
-        if (pat is LitPattern lpat) {
-          if (!alternatives.Exists(x => object.Equals(x.Value, lpat.OptimisticallyDesugaredLit.Value))) {
-            alternatives.Add(lpat.OptimisticallyDesugaredLit);
-          }
+        LiteralExpr lit = null;
+        if (pat is LitPattern lpat) lit = lpat.OptimisticallyDesugaredLit;
+        if (pat is IdPattern id && id.ResolvedLit != null) lit = id.ResolvedLit;
```

Comment:
Enclose the "then" branches with curly braces.

Reply:
Done